### PR TITLE
feat: Transactions topic and schema definition

### PR DESCRIPTION
**Pull request by lynnagara**
I autogenerated this from the example and it's almost certainly wrong - just a starting point we will iterate on.

Kind of wished there was an easy way to share references between topics since transactions has a fair bit of overlap with events.

### CODE REVIEW

**Review by untitaker — APPROVED** [review]
can you just try typing the transactions processor? that'll tell you really quickly what properties are actually required, etc

File: schemas/transactions.v1.schema.json [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+      "type": "object",
+      "title": "transaction_event",
+      "additionalProperties": false,
+      "properties": {
+        "group_id": {
+          "type": "null"
+        },
+        "group_ids": {
+          "type": "array",
+          "maxItems": 0
```
[the remark this answers is not in the document]
Comment (lynnagara) [reply schemas/transactions.v1.schema.json:22]: yeah, i think i had the issue platform feature turned off, let me try and figure out what it should be

**Comment by lynnagara** [comment]
> can you just try typing the transactions processor? that'll tell you really quickly what properties are actually required, etc

I'm being intentionally strict as I think it's easier to loosen later but not go the other way. There was an ask to document the spans part beyond what Snuba actually does anything with so I'm leaning in the direction of more detail in response to that request. Though we may want to loosen that later too.

File: schemas/transactions.v1.schema.json [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+    { "const": "insert" },
+    {
+      "$ref": "#/definitions/TransactionEvent"
+    }
+  ],
+  "definitions": {
+    "TransactionEvent": {
+      "type": "object",
+      "title": "transaction_event",
+      "additionalProperties": false,
```
[the remark this answers is not in the document]
Comment (lynnagara) [reply schemas/transactions.v1.schema.json:15]: We will probably need to, i'm kind of curious to observe in what ways this fails so gonna leave it in this iteration.

**Pull request merged**